inflation/lp/InflationLP.py

```diff
@@ -0,0 +1,1890 @@
+"""
+The module generates the semidefinite program associated to a quantum inflation
+instance (see arXiv:1909.10519).
+
+@authors: Emanuel-Cristian Boghiu, Elie Wolfe, Alejandro Pozas-Kerstjens
+"""
+
+import numpy as np
+import sympy as sp
+
+from collections import Counter, defaultdict
+from gc import collect
+from itertools import chain
+from numbers import Real
+from tqdm import tqdm
+from typing import List, Dict, Tuple, Union, Any
+from warnings import warn
+from scipy.sparse import coo_matrix, vstack
+
+from inflation import InflationProblem
+
+from .numbafied import (nb_apply_lexorder_perm_to_lexboolvecs,
+                        nb_outer_bitwise_or,
+                        nb_outer_bitwise_xor)
+from .writer_utils import write_to_lp, write_to_mps
+
+from ..sdp.fast_npa import nb_is_knowable as is_knowable
+from .monomial_classes import InternalAtomicMonomial, CompoundMoment
+from ..sdp.quantum_tools import (flatten_symbolic_powers,
+                                 party_physical_monomials)
+from .lp_utils import solveLP, solve_Gurobi
+from functools import reduce
+from ..utils import clean_coefficients, eprint, partsextractor, \
+    expand_sparse_vec
+from functools import cached_property
+
+class InflationLP(object):
+    constant_term_name = "constant_term"
+
+    def __init__(self,
+                 inflationproblem: InflationProblem,
+                 nonfanout: bool = False,
+                 supports_problem: bool = False,
+                 default_non_negative: bool = True,
+                 include_all_outcomes: bool = False,
+                 verbose: int = 0) -> None:
+        """
+        Class for generating and solving an LP relaxation for fanout or 
+        nonfanout inflation.
+
+        Parameters
+        ----------
+        inflationproblem : InflationProblem
+            Details of the scenario.
+        nonfanout : bool, optional
+            Whether to consider only nonfanout inflation (GPT problem) or
+            fanout inflation (classical problem). By default ``False``.
+        supports_problem : bool, optional
+            Whether to consider feasibility problems with distributions, or 
+            just with the distribution's support. By default ``False``.
+        default_non_negative : bool, optional
+            Whether to set all variables to be non-negative by default. By
+            default ``True``.
+        include_all_outcomes : bool, optional
```

Comment:
Confirm that this is what include_all_outcomes actually means

Reply:
It is correct.